toml/decoder.py

```diff
@@ -277,11 +277,8 @@ def loads(s, _dict=dict, decoder=None):
                 openstrchar = ""
         if item == '#' and (not openstring and not keygroup and
                             not arrayoftables):
-            j = i
             try:
-                while sl[j] != '\n':
-                    sl[j] = ' '
-                    j += 1
+                beginline = decoder.comment_lint(sl, i, beginline)
```

Comment:
Why can't you use handle comment here?

Reply:
This part of the code just validates the toml file character by character and does not load anything to the retval(current_level) dictionary.